src/main/java/mServer/crawler/RunSender.java

```diff
@@ -39,70 +41,83 @@
     public Date endZeit = new Date(); // zur Sicherheit
 
     public boolean fertig = false;
-    long[] counter = new long[Count.values().length];
-
-    public enum Count {
-
-        ANZAHL("Seiten"), FILME("Filme"), FEHLER("Fehler"), FEHLVERSUCHE("FVers"), WARTEZEIT_FEHLVERSUCHE("ZeitFV[s]"),
-        PROXY("Proxy" /*Seite über Proxy laden*/), NO_BUFFER("NoBuffer"),
-        SUM_DATA_BYTE("sumData" /*Datenmenge, entpackt*/), SUM_TRAFFIC_BYTE("sumTraffic" /*Datenmenge die übertragen wird*/),
-        SUM_TRAFFIC_LOADART_NIX("trNix"), SUM_TRAFFIC_LOADART_DEFLATE("trDeflate"), SUM_TRAFFIC_LOADART_GZIP("trGzip"),
-        GET_SIZE_SUM("getSize"), GET_SIZE_SUM403("gS-403"), GET_SIZE_PROXY("gS-proxy");
-
-        final String name;
-
-        Count(String name) {
-            this.name = name;
-        }
-
-        public static String[] getNames() {
-            String[] ret = new String[values().length];
-
-            for (int i = 0; i < values().length; ++i) {
-                ret[i] = values()[i].name;
-            }
-            return ret;
-        }
-    }
+    public final ConcurrentHashMap<Count, AtomicLong> counter = new ConcurrentHashMap<>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The type of the "counter" object should be an interface such as "ConcurrentMap" rather than the implementation "ConcurrentHashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1319)